### PR TITLE
DOC: Update quick-start instructions

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

_Add a comprehensive description of proposed changes_

_List associated issue number(s) if exist(s): #6 (for example)_

_Documentation PR (if needed): #1340 (for example)_

_Benchmarks PR (if needed): https://github.com/IntelPython/scikit-learn_bench/pull/155 (for example)_

---

The current installation instructions suggest installing sklearnex from the conda-forge channel as the recommended way, but this channel does not come with GPU support per the table. It also mentions that the package can be installed from the main anaconda channel as a third option.

This PR changes the docs so that the conda intel channel would be recommended over the conda-forge one, and removes the instructions for the main anaconda channel, as they do not distribute recent versions of sklearnex ([last release there](https://anaconda.org/main/scikit-learn-intelex) is 2023.1.1)

It also updates the list of supported sklearn versions to include the latest releases (which are now supported), as well as the list of supported python versions to include 3.13, for which there hopefully will be wheels available by the next sklearnex release.

Since right now there is no differentiation of features by OS or python version, it also changes the format of the support tables to something simpler and easier to read.

Besides the support for CPU/GPU, the PR also adds info about other supported devices (FPGAs, which are covered by DPC) and about SPMD mode, which is currently not available in all distribution channels.

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.

**Performance**

Not applicable.

### CODE REVIEW

**Comment by david-cortes-intel** [comment]
Screenshow of how it'd look like after the PR:
image

File: doc/sources/quick-start.rst [hunk 1]
```diff
@@ -370,12 +310,15 @@ Software Requirements
 
          If you use accelerators, refer to `oneAPI DPC++/C++ Compiler System Requirements <https://www.intel.com/content/www/us/en/developer/articles/system-requirements/intel-oneapi-dpcpp-system-requirements.html>`_.
 
-Intel(R) Extension for Scikit-learn is compatible with the last four versions of scikit-learn:
+Intel(R) Extension for Scikit-learn is compatible with the last stable releases of scikit-learn:
```
Comment (icfaust) [inline doc/sources/quick-start.rst:313]: @napetrov @syakov-intel please review this, I'd prefer something a little more ambiguous. "compatible with four major releases of scikit-learn.

**Review by icfaust — COMMENTED** [review]
I'd like for some others to weigh in, as it has an impact on our support burden.

**Comment by david-cortes-intel** [comment]
Closing in favor of a larger PR where these changes were merged: https://github.com/uxlfoundation/scikit-learn-intelex/pull/2282
